aws/resource_aws_guardduty_member.go

```diff
@@ -0,0 +1,121 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/guardduty"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsGuardDutyMember() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGuardDutyMemberCreate,
+		Read:   resourceAwsGuardDutyMemberRead,
+		Delete: resourceAwsGuardDutyMemberDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAwsAccountId,
+			},
+			"detector_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"email": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsGuardDutyMemberCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).guarddutyconn
+	accountID := d.Get("account_id").(string)
+	detectorID := d.Get("detector_id").(string)
+
+	input := guardduty.CreateMembersInput{
+		AccountDetails: []*guardduty.AccountDetail{{
+			AccountId: aws.String(accountID),
+			Email:     aws.String(d.Get("email").(string)),
+		}},
+		DetectorId: aws.String(detectorID),
+	}
+
+	log.Printf("[DEBUG] Creating GuardDuty Member: %s", input)
+	_, err := conn.CreateMembers(&input)
+	if err != nil {
+		return fmt.Errorf("Creating GuardDuty Member failed: %s", err.Error())
+	}
+	d.SetId(fmt.Sprintf("%s:%s", detectorID, accountID))
+
+	return resourceAwsGuardDutyMemberRead(d, meta)
+}
+
+func resourceAwsGuardDutyMemberRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).guarddutyconn
+
+	idParts := strings.Split(d.Id(), ":")
+	accountID := idParts[1]
+	detectorID := idParts[0]
+
+	input := guardduty.GetMembersInput{
+		AccountIds: []*string{aws.String(accountID)},
+		DetectorId: aws.String(detectorID),
+	}
+
+	log.Printf("[DEBUG] Reading GuardDuty Member: %s", input)
+	gmo, err := conn.GetMembers(&input)
+	if err != nil {
+		if isAWSErr(err, guardduty.ErrCodeBadRequestException, "The request is rejected because the input detectorId is not owned by the current account.") {
```

Comment:
😭

Reply:
Yeah its like S3 `NoSuchBucket` in this regard. 😢